### PR TITLE
Correct HEIC orientation for the ping operation (#1233)

### PR DESCRIPTION
**Pull request by rudchenkos**
This is a follow-up for commit 41fa21a2eabdeee0202a5f32c081ee968784c265

The fix for https://github.com/ImageMagick/ImageMagick/issues/1232 was
not effective for the "ping" operation, when the library is ought to
return image properties without decoding the pixel data.

### CODE REVIEW

File: coders/heic.c [hunk 1]
```diff
@@ -330,14 +335,12 @@ static MagickBooleanType ReadHEICImageByID(const ImageInfo *image_info,
...
   (void) SetImageColorspace(image,YCbCrColorspace,exception);
   decode_options=(struct heif_decoding_options *) NULL;
-  option=GetImageOption(image_info,"heic:preserve-orientation");
-  if (IsStringTrue(option) == MagickTrue)
+  if (preserve_orientation == MagickTrue)
     {
       decode_options=heif_decoding_options_alloc();
       decode_options->ignore_transformations=1;
     }
   else
```
Comment (dlemstra) [inline coders/heic.c:343]: Looks like this else statement should be gone?

File: coders/heic.c [hunk 2]
```diff
@@ -308,6 +306,13 @@ static MagickBooleanType ReadHEICImageByID(const ImageInfo *image_info,
   image->depth=8;
   image->columns=(size_t) heif_image_handle_get_width(image_handle);
   image->rows=(size_t) heif_image_handle_get_height(image_handle);
+
+  preserve_orientation = IsStringTrue(GetImageOption(image_info,
```
Comment (dlemstra) [inline coders/heic.c:310]: Could you change this to `preserve_orientation=IsStringTrue`?

File: coders/heic.c [hunk 3]
```diff
@@ -308,6 +306,13 @@ static MagickBooleanType ReadHEICImageByID(const ImageInfo *image_info,
   image->depth=8;
   image->columns=(size_t) heif_image_handle_get_width(image_handle);
   image->rows=(size_t) heif_image_handle_get_height(image_handle);
+
```
Comment (dlemstra) [inline coders/heic.c:309]: We normally don't add these empty lines. Could you remove them?

File: coders/heic.c (hunk 1 above)
Comment (rudchenkos) [reply coders/heic.c]: oeps :-O

**Comment by rudchenkos** [comment]
Hi Dirk,
Thank you for the review,  I have fixed the problems in a fixup commit (to be squashed into the main commit before merging).

**Comment by rudchenkos** [comment]
Thank you!
Happy Balkoningsdag :)

**Comment by dlemstra** [comment]
Jij ook 👍 And thanks for helping us with this.

**Pull request merged**